warp/src/js_exports/raygun.rs

```diff
@@ -11,3 +23,746 @@ impl RayGunBox {
+                        messages: p
+                            .messages()
+                            .iter()
+                            .map(|m| Message::new(m.clone()))
+                            .collect(),
+                        total: p.total(),
+                    })
+                    .collect();
+
+                Self {
+                    variant: MessagesEnum::Page,
+                    value: serde_wasm_bindgen::to_value(&Page { pages, total }).unwrap(),
+                }
+            }
+        }
+    }
+    pub fn variant(&self) -> MessagesEnum {
+        self.variant
+    }
+    pub fn value(&self) -> JsValue {
+        self.value.clone()
+    }
+}
+#[derive(Serialize, Deserialize)]
+pub struct Page {
+    pub pages: Vec<MessagePage>,
+    pub total: usize,
+}
+#[derive(Serialize, Deserialize)]
+pub struct MessagePage {
+    pub id: usize,
+    pub messages: Vec<Message>,
+    pub total: usize,
+}
+#[wasm_bindgen]
+#[derive(Copy, Clone)]
+pub enum MessagesEnum {
+    List,
+    Stream,
+    Page,
+}
+
+#[wasm_bindgen]
+pub struct MessageOptions {
+    inner: raygun::MessageOptions,
+}
+#[wasm_bindgen]
+impl MessageOptions {
+    #[wasm_bindgen(constructor)]
+    pub fn new() -> Self {
+        Self {
+            inner: raygun::MessageOptions::default(),
+        }
+    }
+
+    pub fn set_date_range(&mut self, range: JsValue) {
+        self.inner = self
+            .inner
+            .clone()
+            .set_date_range(serde_wasm_bindgen::from_value(range).unwrap());
+    }
+
+    pub fn set_range(&mut self, range: JsValue) {
+        self.inner = self
+            .inner
+            .clone()
+            .set_range(serde_wasm_bindgen::from_value(range).unwrap());
+    }
+
+    pub fn set_limit(&mut self, limit: u8) {
+        self.inner = self.inner.clone().set_limit(limit);
+    }
+
+    pub fn set_skip(&mut self, skip: i64) {
+        self.inner = self.inner.clone().set_skip(skip);
+    }
+
+    pub fn set_keyword(&mut self, keyword: &str) {
+        self.inner = self.inner.clone().set_keyword(keyword);
+    }
+
+    pub fn set_first_message(&mut self) {
+        self.inner = self.inner.clone().set_first_message();
+    }
+
+    pub fn set_last_message(&mut self) {
+        self.inner = self.inner.clone().set_last_message();
+    }
+
+    pub fn set_pinned(&mut self) {
+        self.inner = self.inner.clone().set_pinned();
+    }
+
+    pub fn set_reverse(&mut self) {
+        self.inner = self.inner.clone().set_reverse();
+    }
+
+    pub fn set_messages_type(&mut self, ty: JsValue) {
+        self.inner = self
+            .inner
+            .clone()
+            .set_messages_type(serde_wasm_bindgen::from_value(ty).unwrap());
+    }
+
+    pub fn date_range(&self) -> JsValue {
+        serde_wasm_bindgen::to_value(&self.inner.date_range()).unwrap()
+    }
+
+    pub fn range(&self) -> JsValue {
+        serde_wasm_bindgen::to_value(&self.inner.range()).unwrap()
+    }
```

Comment:
Would the range produce a value `JsValue`?

Reply:
Unfortunately it does not.

Reply:
hmm ok. For now we can omit these from being exported since I dont believe access to those values in JS would be needed.

Reply:
To clarify, `serde_wasm_bindgen::to_value()` does turn `Range<...>` into a JsValue. 
I just meant that it wouldn't otherwise.

Reply:
Yeah, in fact I'm noticing this is part of MessageOptions which is only used for input from js side. So I will get rid of all the getters, since it only needs setters.

Reply:
done 👍